lib/models/machine_reading.rb

```diff
@@ -1,43 +1,57 @@
   include Logging
 
   def api_format
-    disk_metrics = Hash.new{|h,k|h[k]=[]}
-    nic_metrics = Hash.new{|h,k|h[k]=[]}
-    metrics = Hash.new
-
+    local_platform_remote_id_inventory = PlatformRemoteIdInventory.new
+    infrastructure_prid = local_platform_remote_id_inventory["i:#{self.id[:infrastructure_platform_id]}"]
+    machine_prid = local_platform_remote_id_inventory["#{infrastructure_prid.platform_key}/m:#{self.id[:machine_platform_id]}"]
+    samples = []
     readings.each do |reading|
-      reading[:disk_metrics].each{|dm|
-        disk_metrics[dm['remote_id']] << dm.slice('reading_at', 'usage_bytes', 'read_kilobytes', 'write_kilobytes').merge(reading_at: reading[:reading_at]) } unless ( reading[:disk_metrics].blank? )
-      reading[:nic_metrics].each{|nm|
-        nic_metrics[nm['remote_id']] << nm.slice('transmit_kilobits','receive_kilobits').merge(reading_at: reading[:reading_at]) } unless ( reading[:nic_metrics].blank? )
+      disks_info = []
+      nics_info = []
+      if reading[:disk_metrics].present?
+        disks_info =  reading[:disk_metrics].map do |dm|
+          if local_platform_remote_id_inventory["#{machine_prid.platform_key}/d:#{dm['platform_id']}"]
+            { id: local_platform_remote_id_inventory["#{machine_prid.platform_key}/d:#{dm['platform_id']}"].remote_id,
+              usage_bytes: dm[:usage_bytes],
+              read_bytes_per_second: (dm[:read_kilobytes] * 1000),
+              write_bytes_per_second: (dm[:write_kilobytes] * 1000)
+            }
+          end
+        end.compact
+      end
+      if reading[:nic_metrics].present?
+        nics_info = reading[:nic_metrics].map do |nm|
+          if local_platform_remote_id_inventory["#{machine_prid.platform_key}/n:#{nm['platform_id']}"]
+            { id: local_platform_remote_id_inventory["#{machine_prid.platform_key}/n:#{nm['platform_id']}"].remote_id,
+              receive_bytes_per_second: (nm[:receive_kilobits] * 125),
+              transmit_bytes_per_second: (nm[:transmit_kilobits] * 125)
+            }
+          end
+        end.compact
+      end
+      samples = { start_time: reading[:start_time].iso8601,
+                  end_time: reading[:end_time].iso8601,
+                  machine: reading[:machine_metrics],
+                  nics: nics_info,
+                  disks: disks_info
+                }
     end
-
-    disks = disk_metrics.map{|k,v| {id: k, readings: v } }
-    nics  = nic_metrics.map {|k,v| {id: k, readings: v } }
-
-    machine = readings.map{|reading|
-      { reading_at: reading[:reading_at],
-        cpu_usage_percent: reading[:machine_metrics]['cpu_usage_percent'],
-        memory_bytes: reading[:machine_metrics]['memory_bytes'] } }
-
-    metrics[:readings] = machine unless machine.empty?
-    metrics[:nics] = nics unless nics.empty? # better!! way
-    metrics[:disks] = disks unless disks.empty?
-
-    metrics
+    logger.info "SAMPLES SUBMITTED => #{samples}\n\n"
```

Comment:
should we leave this log information? if so, maybe we may need to make it more explicit to ease debugging process